frontend/Interfaces/buddy/DIP/DIP.h

```diff
@@ -328,12 +328,15 @@ inline MemRef<float, 2> Rotate2D(Img<float, 2> *input, float angle,
 // User interface for 2D Resize.
 inline MemRef<float, 2> Resize2D(Img<float, 2> *input, INTERPOLATION_TYPE type,
                                  std::vector<float> scalingRatios) {
-  if (!scalingRatios[0] || !scalingRatios[1]) {
+  if (scalingRatios[0] <= 0 || scalingRatios[1] <= 0) {
     throw std::invalid_argument(
-        "Please enter non-zero values of scaling ratios.\n"
+        "Please enter positive values of scaling ratios.\n"
         "Note : scaling ratio = "
-        "input_image_dimension / output_image_dimension\n");
+        "output_image_dimension / input_image_dimension\n");
   }
+  std::reverse(scalingRatios.begin(), scalingRatios.end());
```

Comment:
Also adjust the OpenCV dimension(previous PR give [h, w], but it need [w, h]), so the dimension conversion is no longer required.

Reply:
I am not sure I understand, we need to feed dimensions in [h, w] format at the lower level and we capture input in [w, h] format from the higher level.

Reply:
Please change the benchmark code to take input in the [w, h] format. 

Applying `std::reverse` makes it easier to read the code imo, because then we use `scalingRatio[0]` to get `outputSize[0]` and `scalingRatio[1]` to get `outputSize[1]`.

Reply:
> Please change the benchmark code to take input in the [w, h] format.

Please have a review at the [resize2d PR](https://github.com/buddy-compiler/buddy-benchmark/pull/61). The tasks mentioned were in the new commits.